datm/cime_config/buildnml

```diff
@@ -210,9 +202,9 @@ def buildnml(case, caseroot, compname):
 
     # NOTE: User definition *replaces* existing definition.
     files = Files(comp_interface="nuopc")
-    definition_file = [files.get_value("NAMELIST_DEFINITION_FILE", {"component":"datm"})]
+    definition_file = [files.get_value("NAMELIST_DEFINITION_FILE", {"component":compname})]
```

Comment:
Why this change?  This is already component datm.

Reply:
Because its a further step in trying to unify more aspects of buildnml across the data components. I wanted to move this part of buildnml to CDEPS/cime_config - but it was not straightforward to do that. I agree about moving this to a shared location - but the problem is that this routine ends up calling the datm routine _create_namelist - so I was not sure how to accomplish that.